src/viewProviders/flinkArtifacts.ts

```diff
@@ -50,12 +39,52 @@ export class FlinkArtifactsViewProvider
             const loader = CCloudResourceLoader.getInstance();
             this._artifacts = await loader.getFlinkArtifacts(this.computePool!);
           } catch (error) {
-            this.logger.error("Failed to load Flink artifacts", { error });
+            let showNotification = false;
+            let message = "Failed to load Flink artifacts.";
+
+            if (isResponseError(error)) {
+              const status = error.response.status;
+              // Only show notifications for error status codes
+              if (status >= 400 && status < 600) {
+                showNotification = true;
+                switch (status) {
+                  case 401:
```

Comment:
We should also catch the `400`s directly and pass the message `detail` through, since it currently falls back to the default case:
<img width="1617" height="427" alt="Image" src="https://github.com/user-attachments/assets/cf8391b2-6f2d-4507-b4b8-ab931084d23e" />